component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/AuthProxyUtils.java

```diff
@@ -18,127 +18,130 @@
-        SSLSocketFactory sslSocketFactory = getTrustedSSLSocketFactory(properties);
-        TrustManager[] trustManagers = trustManagerFactory.getTrustManagers();
-        if (trustManagers.length != 1 || !(trustManagers[0] instanceof X509TrustManager)) {
-            throw new IllegalStateException("Unexpected default trust managers:" + Arrays.toString(trustManagers));
+        if (proxyProtocol != null) {
+            protocol = proxyProtocol;
         }
-        okHttpClientBuilder = new okhttp3.OkHttpClient.Builder()
-                .sslSocketFactory(sslSocketFactory, (X509TrustManager) trustManagers[0])
-                .proxy(new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyHost, Integer.parseInt(proxyPort))));
-
-        if (StringUtils.isNotEmpty(proxyUsername) && StringUtils.isNotEmpty(proxyPassword)) {
-            Authenticator proxyAuthenticator = (route, response) -> {
-                String credential = Credentials.basic(proxyUsername, proxyPassword);
-                return response.request().newBuilder().header("Proxy-Authorization", credential).build();
-            };
-            okHttpClientBuilder.proxyAuthenticator(proxyAuthenticator);
-        }
-
-        return okHttpClientBuilder.build();
-    }
-
-    private static SSLSocketFactory getTrustedSSLSocketFactory(Map<String, String> configProperties) {
 
+        PoolingHttpClientConnectionManager pool = null;
         try {
-            String keyStorePassword = configProperties.get(Constants.KEYSTORE_PASSWORD);
-            String keyStoreLocation = configProperties.get(Constants.KEYSTORE_LOCATION);
-            String trustStorePassword = configProperties.get(Constants.TRUSTSTORE_PASSWORD);
-            String trustStoreLocation = configProperties.get(Constants.TRUSTSTORE_LOCATION);
-
-            KeyStore keyStore = loadKeyStore(keyStoreLocation, keyStorePassword);
-            trustStore = loadTrustStore(trustStoreLocation, trustStorePassword);
-            return initSSLConnection(keyStore, keyStorePassword);
-        } catch (KeyManagementException | NoSuchAlgorithmException | KeyStoreException | CertificateException
-                | IOException | UnrecoverableKeyException e) {
-            log.error("Error while creating the SSL socket factory", e);
-            return null;
+            pool = getPoolingHttpClientConnectionManager(properties);
+        } catch (HttpClientException e) {
+            log.error("Error while getting http client connection manager", e);
         }
-    }
-
-    private static SSLSocketFactory initSSLConnection(KeyStore keyStore, String keyStorePassword) throws
-            NoSuchAlgorithmException, UnrecoverableKeyException, KeyStoreException, KeyManagementException {
 
-        KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(ALGORITHM);
-        keyManagerFactory.init(keyStore, keyStorePassword.toCharArray());
+        HttpHost host = new HttpHost(proxyHost, proxyPort, protocol);
```

Comment:
cant we just use the proxyProtocol here?

Reply:
You mean create http host as `new HttpHost(protocol);`?

Reply:
fixed